specifyweb/specify/views.py

```diff
@@ -502,9 +510,29 @@ def update_record(record: models.Model):
                     return response
 
     # Dedupe by deleting the agent that is being replaced and updating the old model ID to the new one
-    target_model.objects.get(id=old_model_id).delete()
-
-    return http.HttpResponse('', status=204)
+    for old_model_id in old_model_ids:
+        target_model.objects.get(id=old_model_id).delete()
+
+    # Update new record with json info, if given
+    has_new_record_info: bool = False if new_record_info is None else True
+    if has_new_record_info:
+        obj = api.put_resource(new_record_info['collection'],
+                               new_record_info['specify_user'],
+                               model_name,
+                               new_model_id,
+                               new_record_info['version'],
+                               new_record_info['new_record_data'])
+
+    # Return http response
+    if has_new_record_info:
```

Comment:
should permissions be checked before `.delete()` and `.put_resource()` were called?